cpp/bench/ann/src/common/benchmark.hpp

```diff
@@ -314,17 +314,16 @@ void bench_search(::benchmark::State& state,
 
       queries_processed += n_queries;
     }
+    auto end      = std::chrono::high_resolution_clock::now();
+    auto duration = std::chrono::duration_cast<std::chrono::duration<double>>(end - start).count();
+    if (state.thread_index() == 0) { state.counters.insert({{"end_to_end", duration}}); }
+    state.counters.insert({"Latency", {duration, benchmark::Counter::kAvgIterations}});
```

Comment:
I have also applied Artem's suggestion to store latency values with `benchmark::Counter::kAvgIterations` marker.

Earlier we manually divided by number of iterations and let gbench average over threads using `kAvgThreads`. Since iterations are counted as total iterations performed by all threads, using kAviIterations leads to the same results (without manual divisions by iterations).